internal/resources/policy/kind/quota/input_schema.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policykindquota
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	policyrecipequotamodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/quota"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy"
+	reciperesource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/kind/quota/recipe"
+)
+
+var (
+	inputSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Input for the quota policy, having one of the valid recipes: small, medium, large or custom.",
+		Required:    true,
+		MaxItems:    1,
+		MinItems:    1,
+		ForceNew:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				reciperesource.CustomKey: reciperesource.Custom,
+				reciperesource.SmallKey:  reciperesource.Small,
+				reciperesource.MediumKey: reciperesource.Medium,
+				reciperesource.LargeKey:  reciperesource.Large,
+			},
+		},
+	}
+	RecipesAllowed = [...]string{reciperesource.CustomKey, reciperesource.SmallKey, reciperesource.MediumKey, reciperesource.LargeKey}
+)
+
+type (
+	Recipe string
+	// InputRecipe is a struct for all types of quota policy inputs.
+	inputRecipe struct {
+		recipe      Recipe
+		inputCustom *policyrecipequotamodel.VmwareTanzuManageV1alpha1CommonPolicySpecQuotaV1Custom
```

Comment:
why do you need two attributes here ? 
can't we derive from the recipe type  ?

Reply:
We have two different kinds of recipes (overall,4 - out of which 3 are the empty kinds with pre-defined values), for which we are using input<recipe> to define the spec. We have been using similar approach  in all policies, starting with Security policy as every recipe has a different set of attributes.

Reply:
The third attribute is just being used for assigning nil and does not hold any values, so I believe this is something which can be avoided, Just give a thought

Reply:
This was the only way in which we could have defined the recipes with empty body and not by the recipe name to be in sync with other policies. As per our discussion in the past where I mentioned that we can just take recipe as string and pass nothing inside input, you suggested of not doing it and passing recipe inside input with nothing inside it. Please refer to the examples provided. If we have to remove this, we rather modify the whole quota policy and start give recipe name out of the input scope.

Reply:
I remember the discussion. 
The present schema defined looks good and would not need any change. 

This comment is based on the coding. 
Even if you have 4 recipes, 
The inputCustom and InputCommon are mutually exclusive to each other. 

If you see the values assigned. 
inputCommon is assigned as nil when InputCustom is not being used

Reply:
This would tie the logic very closely with the existing policy recipes. It's just a thought, but I think it would cause tight coupling.

I believe the current code is more readable and extensible. But feel free to update if you feel strongly about it.

Reply:
I have taken a single variable- `input` and used it both for custom and the pre-defined template recipes.